pkg/cmd/pr/edit/edit_test.go

```diff
@@ -530,16 +542,107 @@ func Test_editRun(t *testing.T) {
 				},
 				Fetcher: testFetcher{},
 			},
-			httpStubs: func(reg *httpmock.Registry) {
-				mockRepoMetadata(reg, false)
+			httpStubs: func(reg *httpmock.Registry, t *testing.T) {
+				mockRepoMetadata(reg, mockRepoMetadataOptions{reviewers: true, teamReviewers: false, assignees: true, labels: true, projects: true, milestones: true})
 				mockPullRequestUpdate(reg)
-				mockPullRequestReviewersUpdate(reg)
+				mockPullRequestRemoveReviewers(reg)
 				mockPullRequestUpdateLabels(reg)
 				mockPullRequestUpdateActorAssignees(reg)
 				mockProjectV2ItemUpdate(reg)
 			},
 			stdout: "https://github.com/OWNER/REPO/pull/123\n",
 		},
+		// Conditional team fetching cases
+		{
+			name: "non-interactive add only user reviewers skips team fetch",
+			input: &EditOptions{
+				Detector:    &fd.EnabledDetectorMock{},
+				SelectorArg: "123",
+				Finder:      shared.NewMockFinder("123", &api.PullRequest{URL: "https://github.com/OWNER/REPO/pull/123"}, ghrepo.New("OWNER", "REPO")),
+				Interactive: false,
+				Editable: shared.Editable{
+					Reviewers: shared.EditableSlice{Add: []string{"monalisa", "hubot"}, Edited: true},
+				},
+				Fetcher: testFetcher{},
+			},
+			httpStubs: func(reg *httpmock.Registry, t *testing.T) {
+				// reviewers only (users), no team reviewers fetched
+				mockRepoMetadata(reg, mockRepoMetadataOptions{reviewers: true})
+				// explicitly assert that no OrganizationTeamList query occurs
+				reg.Exclude(t, httpmock.GraphQL(`query OrganizationTeamList\b`))
+				mockPullRequestUpdate(reg)
+				mockPullRequestAddReviewers(reg)
+			},
+			stdout: "https://github.com/OWNER/REPO/pull/123\n",
+		},
+		{
+			name: "non-interactive add contains team reviewers skips team fetch",
+			input: &EditOptions{
+				Detector:    &fd.EnabledDetectorMock{},
+				SelectorArg: "123",
+				Finder:      shared.NewMockFinder("123", &api.PullRequest{URL: "https://github.com/OWNER/REPO/pull/123"}, ghrepo.New("OWNER", "REPO")),
+				Interactive: false,
+				Editable: shared.Editable{
+					Reviewers: shared.EditableSlice{Add: []string{"monalisa", "OWNER/core"}, Edited: true},
+				},
+				Fetcher: testFetcher{},
+			},
+			httpStubs: func(reg *httpmock.Registry, t *testing.T) {
+				// reviewer add includes team but non-interactive Add/Remove provided -> no team fetch
+				mockRepoMetadata(reg, mockRepoMetadataOptions{reviewers: true})
+				reg.Exclude(t, httpmock.GraphQL(`query OrganizationTeamList\b`))
```

Comment:
Maybe copy/paste the same comment from the previous test case:


```suggestion
				// explicitly assert that no OrganizationTeamList query occurs
				reg.Exclude(t, httpmock.GraphQL(`query OrganizationTeamList\b`))
```